plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -216,29 +216,25 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni DefaultFun where
        makeBuiltinMeaning
             (\b x y -> if b then x else y)
             (runCostingFunThreeArguments . paramIfThenElse)
-    toBuiltinMeaning CharToString =
-        makeBuiltinMeaning
-            (pure :: Char -> String)
-            mempty  -- TODO: budget.
     toBuiltinMeaning Append =
         makeBuiltinMeaning
-            ((++) :: String -> String -> String)
+            ((<>) :: Text -> Text -> Text)
             mempty  -- TODO: budget.
     toBuiltinMeaning EqualsString =
         makeBuiltinMeaning
-            ((==) @String)
+            ((==) @Text)
             mempty  -- TODO: budget.
     toBuiltinMeaning EncodeUtf8 =
         makeBuiltinMeaning
-            (BSC.pack :: String -> BS.ByteString)
+            (encodeUtf8 :: Text -> BS.ByteString)
             mempty  -- TODO: budget.
     toBuiltinMeaning DecodeUtf8 =
         makeBuiltinMeaning
-            (BSC.unpack :: BS.ByteString -> String)
+            (decodeUtf8 :: BS.ByteString -> Text)
```

Comment:
I see that the [documentation](https://hackage.haskell.org/package/text-1.2.5.0/docs/Data-Text-Encoding.html#v:decodeUtf8) for `decodeUtf8` says "Decode a ByteString containing UTF-8 encoded text that is known to be valid.  **If the input contains any invalid UTF-8 data, an exception will be thrown that cannot be caught in pure code.**".  We ran into something like this yesterday, and I think that could maybe crash the evaluator on the chain, which we definitely don't want.  There seem to be a couple of safer operations, but it's still not clear to me if they'd solve this.

Reply:
Great catch. We should use https://hackage.haskell.org/package/text-1.2.5.0/docs/Data-Text-Encoding.html#v:decodeUtf8-39- and have an evaluation failure if it's Left. @ak3n could you do that?

Reply:
Yeah, sure.

Reply:
It seems that might be there should be an universal handler that catches all the exceptions and wraps with `EvaluationResult a`? Maybe inside `toBuiltinMeaning` or somehow else to hide the details.

Reply:
I'm not sure. I think in the past @effectfully suggested that we just catch all exceptions at the top level, but this is hard to do properly without e.g. swallowing async exceptions that might get thrown while you're evaluating. Better to just not throw exceptions that you don't know about.

Reply:
Since then I reeducated myself on how this is done properly, quoting [Catching all exceptions](https://www.schoolofhaskell.com/user/snoyberg/general-haskell/exceptions/catching-all-exceptions#real-solution--separate-worker-thread):

> The concept is simple: if you have some function you want to catch all exceptions for, fork a new thread and run the function there. Catch all exceptions thrown in that new thread, and return them to the original thread (via usage of software transactional memory). Now, if any async exceptions are thrown to the original thread, they are unaffected by the exception catching code. And the cool part that the async library took care of automatically: if the original thread gets an async exception, automatically propagate it down to the worker thread so that it terminates work immediately.